src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/OpenTelemetry.Contrib.Instrumentation.ElasticsearchClient.csproj

```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <Description>Elasticsearch instrumentation for OpenTelemetry .NET</Description>
+    <PackageTags>$(PackageTags);distributed-tracing</PackageTags>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="System.Text.Json" Version="4.7.2" />
```

Comment:
this is so amazing! At first I was like "where is the reference to OpenTelemetry.Api????" But then realized it's not needed any longer.

Reply:
does the project need a newer version of DiagnosticsSource package?

Reply:
Not sure if the amazing was for this PR or that the csproj doesn't have a reference to OpenTelemetry in it. :-)

I think these projects should rely on the version of DiagnosticsSource included with the reference from OpenTelemetry.

Reply:
Something is missing.. I don't know how this project compiles without adding reference to OpenTelemetry project.....

Reply:
we do that using the props/targets file. there we have the version.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/ff5d81fc3695b8d15b56ada54641139cbb3e84be/src/Directory.Build.targets#L9

Reply:
![](https://media2.giphy.com/media/12NUbkX6p4xOO4/giphy.gif)

Reply:
@cijothomas I wonder if we need the full SDK when creating an instrumentation library? Woulddn't API be enough? Or ideally just a DiagnosticsSource.

Let's build a real deal, not the magic trick =).

@ejsmith this comment is not blocking the PR.

Reply:
Its technically feasible to write this without any Otel dependency... as all we need is DiagnosticSource.
In general, instrumentations only need DiagnosticSource dependency.

Reply:
> Its technically feasible to write this without any Otel dependency... as all we need is DiagnosticSource.
> In general, instrumentations only need DiagnosticSource dependency.

do we have an example in either this or main repository?

Reply:
I don't think we can do that yet. For example, we are adding an extension for `TracerProviderBuilder` and that is only available from OpenTelemetry. What we could do is move all things related to instrumentation/exporters to another simplified package and, those projects (instrumentation/exporters) would need that.

Reply:
should we create an issue in the main repo to discuss it? I think this is a nice goal to simplify the dependencies before 1.0 release

Reply:
We are still reviewing the API/SDK things to simplify this. What we could do is create a 0.1-beta (something like this), just for people to test and see. And, later, we can update those to be compliant with api only.

Reply:
but, sure, we can open an issue! :)

Reply:
Yes, this is something we are trying to achieve. I think there is issue opened for this. If not I can create.

I tried to clarify that Otel dependency is not required here:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/examples/Console/TestConsoleExporter.cs#L38-L39
But its not very obvious to most. 
The to-be-written document about "How to instrument your library" will cover this.

Reply:
I guess it does seem like it would be useful to have generic adaptor projects with no dependencies on otel that convert from diagnosticsource to activitysource, but it feels like it would be very silly for a small library like this to have to have a project for the generic adaptor and then another one that integrates it with otel (adds builder extension methods). Ideally, library authors would implement ActivitySource's on their own and these adaptors wouldn't be needed at all. But in this particular instrumentation, I'm using the SuppressInstrumentationScope which would still require an otel dependency for the activitysource adapter.